virtcontainers/qemu.go

```diff
@@ -1290,7 +1313,34 @@ func (q *qemu) addDevice(devInfo interface{}, devType deviceType) error {
 
 	switch v := devInfo.(type) {
 	case types.Volume:
-		q.qemuConfig.Devices = q.arch.append9PVolume(q.qemuConfig.Devices, v)
+		if q.config.SharedFS == config.VirtioFS {
+			q.Logger().WithField("volume-type", "virtio-fs").Info("adding volume")
+
+			randBytes, err := utils.GenerateRandomBytes(8)
+			if err != nil {
+				return err
+			}
+			id := hex.EncodeToString(randBytes)
+
+			sockPath, err := utils.BuildSocketPath(store.RunVMStoragePath, q.id, "vhost-fs.sock")
```

Comment:
is this the same socket path used by the fs daemon?  if so, I think you should implement a new function to make sure the same path is returned in both cases.

Reply:
Will fix in the next revision.